src/libraries/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -65,10 +65,7 @@ public partial struct JsonArrayEnumerator : System.Collections.Generic.IEnumerat
         public bool MoveNext() { throw null; }
         void System.Collections.IEnumerator.Reset() { }
     }
-    public sealed partial class JsonBoolean : System.Text.Json.JsonNode,
-#nullable disable // to enable use with both T and T? for reference types due to IEquatable<T> being invariant
-        System.IEquatable<System.Text.Json.JsonBoolean>
-#nullable restore
```

Comment:
Can you help me understand this? Why did we disabled nullability before and why can we re-enable it now? What changed?

Reply:
I don't know much, by my understanding `IEquatable<T>` had issue with nullable reference type parameter and we did this workaround, but that is fixed recently https://github.com/dotnet/roslyn/pull/37215 and reverted back the workaround